### PR TITLE
New `data_source_role` resource

### PR DESCRIPTION
**Pull request by skatsaounis**
Adds new `data_source_role ` resource which fetches RBAC roles by name.

It can solve https://github.com/grafana/terraform-provider-grafana/issues/700 as well.

Relevant Go Client PR: https://github.com/grafana/grafana-api-golang-client/pull/151

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -72,10 +72,6 @@ github.com/google/uuid v1.3.0 h1:t6JiXgmwXMjEs8VusXIJk2BXHsn+wx8BZdTaoZ5fu7I=
 github.com/google/uuid v1.3.0/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
 github.com/grafana/amixr-api-go-client v0.0.7 h1:U6W6yKxMMybI+Qz4zl+Vih48o6CczLaU/vjk2m7omvU=
 github.com/grafana/amixr-api-go-client v0.0.7/go.mod h1:N6x26XUrM5zGtK5zL5vNJnAn2JFMxLFPPLTw/6pDkFE=
-github.com/grafana/grafana-api-golang-client v0.21.0 h1:PQ2Wfo9jMMiftC4VRMlJxbUNvYCXMV1YFDKm7Ny3SaM=
```
Comment (skatsaounis) [inline go.sum]: Will revert when Go client PR is merged

File: go.mod [hunk 2]
```diff
@@ -76,3 +76,5 @@ require (
 	google.golang.org/grpc v1.54.0 // indirect
 	google.golang.org/protobuf v1.29.1 // indirect
 )
+
```
Comment (skatsaounis) [inline go.mod]: Will revert when Go client PR is merged

**Comment by skatsaounis** [comment]
@inkel I update the PR with the new client version and it is ready to be reviewed.

**Review by inkel — CHANGES_REQUESTED** [review]
CI is failing because it's missing the `go.sum` updated entry for the Grafana Go client. By looking at the list of changed files it seems you forgot to include it in your last commit, please add it and I'll happily review the PR.

**Comment by skatsaounis** [comment]
@inkel I think we are good to go now. I tried to rebase and fix conflicts and somehow I messed up `go.sum`. Please take a look

**Review by julienduchesne — APPROVED** [review]
LGTM!

**Pull request merged**